api/subscriptions/subscriptions.go

```diff
@@ -202,29 +205,29 @@ func (s *Subscriptions) handleSubject(w http.ResponseWriter, req *http.Request)
 		return utils.HTTPError(errors.New("not found"), http.StatusNotFound)
 	}
 
-	conn, closed, err := s.setupConn(w, req)
+	conn, closed, subject, err := s.setupConn(w, req)
 	// since the conn is hijacked here, no error should be returned in lines below
 	if err != nil {
```

Comment:
lets double check if it does not hijack the ws connection like last time :D

Reply:
Ya I tested making WS connections with this as I wanted to see that the metrics worked